### ISSUE TITLE
Fetching denormalized grapher link ignores metadata

### ISSUE THREAD

**Issue by Floriferous** [cult-of-coders/grapher#367]
If you have a metadata link, and you also denormalized the link, its metadata will not be fetched.

For example:
```
const A = new Mongo.Collection('A');
const B = new Mongo.Collection('A');

A.addLinks({
  b: { field: 'bLinks', collection: B, metadata: true, type: 'many' },
});

B.addLinks({
  a: {
    inversedby: 'b',
    collection: A,
    denormalize: {
      field: 'aCache',
      body: { title: 1 },
    },
  },
});

const bId = B.insert({});
A.insert({
  bLinks: [{ _id: bId, meta: 'hello world' }],
  title: 'yo',
});

B.fetchOne({ $filters: { _id: bId }, title: 1 });
// Query is broken, as it does not return the metadata, because the title is fetched through
// the denormalized part.
// Only if you try to query a non-denormalized field will the metadata appear again.
```

I expect metadata to always be present when I do grapher fetches, otherwise it gets difficult to manage, as you'd have to always fetch a non-denormalized field if you need access to the metadata, which is convoluted and unclear.

**Comment by theodorDiaconu** [comment]
This is so complicated to solve @Floriferous . I have to change a lot of things. Also, I really dislike making any fixes for metadata since they were bad design and if I would code Grapher again, I would simply remove this ability. 

Can you refactor so you have an intermediary collection?

**Comment by theodorDiaconu** [comment]
Maybe stating it's complicated was a stretch, it's not that complicated, but it pollutes the code in a way I would like to avoid:

Steps:
1. When we detect an inversed meta caching going on, we also have to cache the full links. Might require some changes in denormalize
2. When we do data-fetching we have to also include that field, but schedule it for deletion after metadata
3. When we assemble meta-data we have to be aware of these type of linking.

**Comment by Floriferous** [comment]
Fair enough. I believe you should mention some of these caveats, because this behavior is not easy to understand, and you should be able to count on grapher to return the right data all the time. We're adding a lot of tests because we're not sure if grapher does what it is supposed to do, as we push it to its limits.

For our use-case, I guess that defining all caches outside of grapher might be the better way forward, that way we can explicitly request the caches with their cacheField (which also makes it clear and safe that we want the cached data), and we skip the built-in helpers from grapher.
